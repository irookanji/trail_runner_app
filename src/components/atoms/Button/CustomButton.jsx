import React from "react";
import StyledButton from "./styles.js";

export default function CustomButton(props) {
  return (
    <StyledButton
      textColor={props.textColor}
      customWidth={props.customWidth}
      mTop={props.mTop}
      mBottom={props.mBottom}
      bgColor={props.bgColor}
      hoverColor={props.hoverColor}
    >
      {props.text}
    </StyledButton>
  );
}
