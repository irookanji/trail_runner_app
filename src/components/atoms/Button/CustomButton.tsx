import React from 'react';
import { StyledButton, StyledLink } from './styles';

type Props = {
  link: string; 
  $xsCustomWidth?: string; 
  $buttonWidth?: string;
  $mTop?: string;
  $mBottom?: string;
  white?: boolean | string;
  onClick?: () => void;
  text?: string;
  $textColor?: string;
  $bgColor?: string;
  $hoverColor?: string;
  $customWidth?: string;
}

export default function CustomButton({link, $xsCustomWidth, $buttonWidth, $mTop, $mBottom, white, onClick, text, $customWidth}: Props) {
  return (
    <StyledLink to={link}>
      <StyledButton
        $xsCustomWidth={$xsCustomWidth}
        $customWidth={$buttonWidth}
        $mTop={$mTop}
        $mBottom={$mBottom}
        white={white}
        onClick={onClick}
      >
        {text}
      </StyledButton>
    </StyledLink>
  );
}
