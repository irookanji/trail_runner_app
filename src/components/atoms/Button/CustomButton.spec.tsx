/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import CustomButton from './CustomButton';

test('Button rendered as expected', () => {
  const component = renderer.create(
    <Router>
      <CustomButton link="/" text="Button" />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button rendered as expected with props params', () => {
  const params = {
    link: '/',
    white: true,
  };
  const component = renderer.create(
    <Router>
      <CustomButton {...params}>Button text</CustomButton>
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
