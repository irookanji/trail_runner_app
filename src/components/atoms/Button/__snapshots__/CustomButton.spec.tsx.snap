// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendered as expected 1`] = `
<a
  className="sc-gsDKAQ iAoPo"
  href="/"
  onClick={[Function]}
>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root sc-bdvvtL kFahnm css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Button
  </button>
</a>
`;

exports[`Button rendered as expected with props params 1`] = `
<a
  className="sc-gsDKAQ iAoPo"
  href="/"
  onClick={[Function]}
>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root sc-bdvvtL jePCiI css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    white={true}
  >
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</a>
`;
